docs/bucket/lifecycle/README.md

```diff
@@ -129,9 +129,31 @@ of objects under the prefix `user-uploads/` as soon as there are more than `N` n
     ]
 }
 ```
-
 Note: This rule has an implicit zero NoncurrentDays, which makes the expiry of those 'extra' noncurrent versions immediate.
 
+#### 3.2.b Automatic removal of all versions (MinIO only extension)
+
+This is available only on MinIO as an extension to the Expiration feature. The following rule makes it possible to remove all versions of an object under 
+the prefix `user-uploads/` as soon as the latest object satisfies the expiration criteria.
+
+```
+{
+    "Rules": [
+        {
+            "ID": "Purge all versions of an expired object",
+            "Status": "Enabled",
+            "Filter": {
+                "Prefix": "users-uploads/"
+            },
+            "Expiration": {
+                "Days": 7,
+                "ExpiredObjectAllVersions": true
```

Comment:
The name is a bit weird. How about a more straightforward one:
```suggestion
                "ExpireAllObjectVersions": true
```

Reply:
That naming is because of AWS style naming ExpiredObjectDeleteMarker 

Meaning to indicate delete "Expired" ones. For being somewhat close to their definition.

Reply:
I had named it `ExpireAllObjectVersions` but moved to be same as `ExpiredObjectDeleteMarker`

Reply:
In AWS terminology they read it like this 

```
Expire -> in Days -> ExpiredObjectDeleteMarker
```

So it follows here 

```
Expire -> in Days -> ExpiredObjectAllVersions
```